package-manifest-spec.md

```diff
@@ -0,0 +1,156 @@
+# Package Manifest Specification
+
+This document defines the specification for the **Package Manifest**.  The
+package manifest contains meta data about the package as well as dependencies
+needed for installation of the package.
+
+
+## Guiding Principles
+
+The package manifest specification makes the following assumptions about the
+document lifecycle.
+
+1. Package manifests are long lived documents that will be modified regularly
+2. Package manifests will primarily be consumed by package managers.
+3. Package manifests will primarily be used by package indexes to popululate basic package metadata.
+2. Package Managers will typically be used to assist in the creation and modification of Package Manifest documents.
+4. Package Manifest documents will occasionally be created and modified manually by people.
+4. Package manifests will typically be stored alongised the package source code.
+
+
+## Format
+
+The canoonical format for the package manifest is a JSON document containing a
+single JSON object.  
+
+
+## Filename
+
+Convention is to name this document `epm.json` which is short for *'ethereum
+package manifest'*.
+
+
+## Document Specification
+
+The following fields are defined for the package manifest.  Custom fields may
+be included.  Custom fields **should** be prefixed with `x-` to prevent name
```

Comment:
**must**?

It is probably worth explaining the process of adding fields to the specification.